gsy_e_sdk/setups/Asset_API_scripts/rest_bid_offer_posting.py

```diff
@@ -0,0 +1,147 @@
+
+    def on_tick(self, tick_info):
+        pass
+
+    def post_bid_offer(self):
+        """Post a bid or an offer to the exchange."""
+        for area_uuid, area_dict in self.latest_grid_tree_flat.items():
+            asset_info = area_dict.get("asset_info")
+            if not asset_info:
+                continue
+
+            # Consumption assets
+            required_energy = asset_info.get("energy_requirement_kWh")
+            if required_energy:
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=required_energy
+                )
+
+            # Generation assets
+            available_energy = asset_info.get("available_energy_kWh")
+            if available_energy:
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=available_energy
+                )
+
+            # Storage assets
+            buy_energy = asset_info.get("energy_to_buy")
+            if buy_energy:
+                self.add_to_batch_commands.bid_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=buy_energy
+                )
+
+            sell_energy = asset_info.get("energy_to_sell")
+            if sell_energy:
+                self.add_to_batch_commands.offer_energy_rate(
+                    asset_uuid=area_uuid, rate=10, energy=sell_energy
+                )
+
+            self.execute_batch_commands()
+
+    def on_event_or_response(self, message):
+        pass
+
+    def on_finish(self, finish_info):
+        self.is_finished = True
+
+
+def get_assets_name(node: dict) -> dict:
+    """
+    Parse the grid tree and return all registered assets
+    wrapper for _get_assets_name
+    """
+    if node == {}:
+        return {}
+    reg_assets = {"Area": [], "Load": [], "PV": [], "Storage": []}
+    _get_assets_name(node, reg_assets)
+    return reg_assets
+
+
+def _get_assets_name(node: dict, reg_assets: dict):
+    """
+    Parse the Collaboration / Canary Network registry
+    Return a list of the Market nodes the user is registered to
+    """
+    if node.get("registered") is True:
+        area_type = node["type"]
+        reg_assets[area_type].append(node["name"])
+    for child in node.get("children", []):
+        _get_assets_name(child, reg_assets)
+
+
+aggregator = Oracle(aggregator_name=ORACLE_NAME)
+simulation_id = os.environ["API_CLIENT_SIMULATION_ID"]
+domain_name = os.environ["API_CLIENT_DOMAIN_NAME"]
+websockets_domain_name = os.environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"]
+asset_args = {"autoregister": False, "start_websocket": False}
+if AUTOMATIC:
+    registry = aggregator.get_configuration_registry()
+    registered_assets = get_assets_name(registry)
+    LOAD_NAMES = registered_assets["Load"]
+    PV_NAMES = registered_assets["PV"]
+    STORAGE_NAMES = registered_assets["Storage"]
+
+
+def register_asset_list(asset_names, asset_params, asset_uuid_map):
```

Comment:
Please also add type hints here:
```suggestion
def register_asset_list(asset_names: List, asset_params: Dict, asset_uuid_map: Dict) -> Dict:
```